### PR TITLE
Fix overload resolution for pragmas evaluation

### LINKED ISSUE #4384: Macros not subjected to overload resolution when used as pragmas
``` nim
macro testMacro(body: untyped): stmt = discard
macro testMacro(s: string, body: untyped): stmt = discard
proc foo() {.testMacro: "foo".}
```

```
test.nim(3, 1) Error: expression 'proc foo() {..}' has no type (or is ambiguous)
```

Change `testMacro` definition order, and it works.

### LINKED ISSUE #6448: Local macro overrides imported macro silently
```nim
import asyncdispatch, macros

macro async*(body: untyped): untyped =
  return newStmtList()

proc bar() {.async.} =
  echo 42

proc foo() {.async.} = 
  await bar()

asyncCheck foo()
runForever()
```

Is this behaviour expected? The ``async`` macro in the module appears to override the imported one from ``asyncdispatch``.

### PR DESCRIPTION
**Pull request by LemonBoy**
Fixes #6448
Fixes #4384

Slightly more verbose than I wanted.
**Edit** I've managed to mistype the ticket number at least 3-4 times

### CODE REVIEW

File: compiler/wordrecg.nim [hunk 1]
```diff
@@ -86,6 +86,8 @@ type
     wInOut, wByCopy, wByRef, wOneWay,
     wBitsize,
 
+    wBenign, wRtl, wNilError, wInl, wCompilerRtl
```
Comment (Araq) [inline compiler/wordrecg.nim]: Huh? Why are these required?

Comment (LemonBoy) [reply compiler/wordrecg.nim]: Because we now check all the pragmas and I didn't want to clean up the unused pragmas scattered all around :)

**Comment by LemonBoy** [comment]
> Is the fact that we now have an overloading resolution that keeps silent over non-existing procs an implementation detail or something that needs to be covered by the spec?

I think that the only interesting part that needs to be spec'd is the order of evaluation for `proc` pragmas (macros are evaluated first and in their declaration order, templates and builtin are evaluated later) and the "special" macro rule for `proc`s itself.

File: compiler/wordrecg.nim (hunk 1 above)
Comment (Araq) [reply compiler/wordrecg.nim]: Well I don't see why these custom pragmas need to become builtins.

Comment (LemonBoy) [reply compiler/wordrecg.nim]: I assumed those once were builtins that got removed from the compiler but not from the stdlib, if that's not the case I can either remove them or make them pragma templates.

Comment (Araq) [reply compiler/wordrecg.nim]: https://github.com/nim-lang/Nim/blob/devel/lib/system/inclrtl.nim

**Pull request merged**